Sources/UDP/UDPSocket.swift

```diff
@@ -0,0 +1,114 @@
+// UDPSocket.swift
+//
+// The MIT License (MIT)
+//
+// Copyright (c) 2015 Zewo
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in all
+// copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+
+import Core
+import CLibvenice
+import POSIX
+
+public enum UDPError: Error {
+    case didSendDataWithError(error: SystemError, remaining: Data)
+    case didReceiveDataWithError(error: SystemError, received: Data)
+}
+
+extension UDPError: CustomStringConvertible {
+    public var description: String {
+        switch self {
+        case .didSendDataWithError(let error, _): return "\(error)"
+        case .didReceiveDataWithError(let error, _): return "\(error)"
+        }
+    }
+}
+
+public final class UDPSocket {
+    private var socket: udpsock?
+    public private(set) var closed = false
+
+    public var port: Int {
+        return Int(udpport(socket))
+    }
+
+    internal init(socket: udpsock) {
+        self.socket = socket
+    }
+
+    public convenience init(ip: IP) throws {
```

Comment:
Maybe instead of receiving the `IP` we could have initializers with `host` and `port` and then we initialize the `IP` ourselves like we do on `TCP`? What do you think?